### PR TITLE
[KYUUBI #3306][INFRA] Add CI workflow for Kyuubi-AuthZ

### PR DESCRIPTION
**Pull request by Yikf**
### _Why are the changes needed?_

Fix https://github.com/apache/incubator-kyuubi/issues/3306

Currently, Kyuubi-AuthZ support Apache Spark 3.0 and above, and there are no CI workflow to check compatibility with Apache Spark 3.0, this pr aim to add a CI workflow for AuthZ module

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -97,6 +97,53 @@ jobs:
...
+          - '3.0'
+        spark-archive: [""]
+        exclude-tags: [""]
+        comment: ["normal"]
+        include:
+          - java: 8
+            spark: '3.0'
+            spark-archive: '-Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.0.0 -Dspark.archive.name=spark-3.0.0-bin-hadoop3.2.tgz'
+            exclude-tags: '-Dmaven.plugin.scalatest.exclude.tags=org.scalatest.tags.Slow'
+            comment: 'verify-on-spark-3.0-binary'
```
Comment (pan3793) [inline .github/workflows/master.yml:119]: remove this block

File: extensions/spark/kyuubi-spark-authz/pom.xml [hunk 2]
```diff
@@ -288,4 +288,14 @@
         </testResources>
     </build>
 
+    <profiles>
+        <profile>
+            <id>spark-3.0</id>
```
Comment (cfmcgrady) [inline extensions/spark/kyuubi-spark-authz/pom.xml:293]: spark 3.0 is EOL after #2974 . Maybe we can use a command line argument to specify the Spark version. WDYT? cc @pan3793

Comment (pan3793) [reply extensions/spark/kyuubi-spark-authz/pom.xml:293]: sgtm

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -97,6 +97,47 @@ jobs:
...
+    name: Kyuubi-AuthZ and Spark Test
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        java:
+          - 8
+          - 11
+        spark:
+          - '3.0'
```
Comment (pan3793) [inline .github/workflows/master.yml:110]: 3.0.3

File: .github/workflows/master.yml [hunk 4]
```diff
@@ -97,6 +97,47 @@ jobs:
...
+        with:
+          distribution: zulu
+          java-version: ${{ matrix.java }}
+          cache: 'maven'
+          check-latest: false
+      - name: Build and test Kyuubi AuthZ with supported Spark versions
+        run: |
+          TEST_MODULES="extensions/spark/kyuubi-spark-authz"
+          ./build/mvn clean test ${MVN_OPT} -pl ${TEST_MODULES} -am \
+          -Dspark.version=3.0.0
```
Comment (pan3793) [inline .github/workflows/master.yml:131]: -Dspark.version=${{ matrix.spark }}

**Review by pan3793 — APPROVED** [review]

File: .github/workflows/master.yml [hunk 5]
```diff
@@ -97,6 +97,47 @@ jobs:
...
+    strategy:
+      fail-fast: false
+      matrix:
+        java:
+          - 8
+          - 11
+        spark:
+          - '3.0.3'
+        spark-archive: [""]
+        exclude-tags: [""]
```
Comment (cfmcgrady) [inline .github/workflows/master.yml:112]: nit:
```diff
-        spark-archive: [""]
-        exclude-tags: [""]
```

**Review by cfmcgrady — APPROVED** [review]

**Comment by yaooqinn** [comment]
we can separate it to a new yml, and fully test all supported versions in it

**Comment by yaooqinn** [comment]
can we add a maven profile for authz module to enable it when needed?

**Comment by yaooqinn** [comment]
Sorry, I don't mean a profile for spark.version, but a profile for authz

**Comment by yaooqinn** [comment]
it's ok

**Review by yaooqinn — APPROVED** [review]

**Pull request closed without merging**